src/functions.jl

```diff
@@ -110,3 +97,33 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     data = append!(parent(A), unname(B))
     return NamedDimsArray{newL}(data)
 end
+
+################################################
+# Generators
+
+if VERSION > v"1.1-"
+    Base.eachslice(A::NamedDimsArray; dims) = _eachslice(A, dims)
+else
+    eachcol(A::AbstractVecOrMat) = (view(A, :, i) for i in axes(A, 2))
+    eachrow(A::AbstractVecOrMat) = (view(A, i, :) for i in axes(A, 1))
+    # every line identical to Base, but no _eachslice(A, dims) to disatch on.
```

Comment:
@oxinabox previously objected to copying Base code here, so I'll defer to him on this

Reply:
Other approaches are possible, I see that Compat.jl does have eachslice etc. I don't know what you think of depending on that.

Reply:
> I see that Compat.jl does have eachslice etc. I don't know what you think of depending on that.

Now that Compat.jl is for Julia v1.0+ I'd be happy to use it, rather than copy code here :)